internal/pkg/manifest/validate.go

```diff
@@ -252,6 +247,11 @@ func (b BackendServiceConfig) validate() error {
 	if err = b.Network.validate(); err != nil {
 		return fmt.Errorf(`validate "network": %w`, err)
 	}
+	if b.Network.Connect.Alias != nil {
+		if b.RoutingRule.GetTargetContainer() == nil && b.ImageConfig.Port == nil {
```

Comment:
This might be of interest to @paragbhingre.
If we do the changes here: https://github.com/aws/copilot-cli/pull/4192#discussion_r1026933083

Then we can replace this section with `if len(manifest.ExposedPorts()) > 0` and that would greatly simplify things I think